lib/web/controllers/submission_controller.ex

```diff
@@ -22,6 +22,8 @@ defmodule Web.SubmissionController do
 
   plug Web.Plugs.FetchPage when action in [:index, :show, :managed_submissions]
 
+  plug(Web.Plugs.FetchSubmission, [submission_id: "id"] when action in [:edit])
```

Comment:
Switch to this since we're not using the passed in `submission_id`

```suggestion
  plug(Web.Plugs.FetchSubmission when action in [:edit])
```